### PR TITLE
Ignore charset encoding in header

### PR DESCRIPTION
**Pull request by jefioliveira**
I need to ignore the charset in the header.

Issue #15
issue #225

### CODE REVIEW

**Comment by jefioliveira** [comment]
I signed it!

File: lib/src/base_client.dart [hunk 1]
```diff
@@ -153,20 +153,23 @@ abstract class BaseClient implements Client {
...
-    if (body != null) {
-      if (body is String) {
-        request.body = body;
-      } else if (body is List) {
-        request.bodyBytes = body.cast<int>();
-      } else if (body is Map) {
-        request.bodyFields = body.cast<String, String>();
-      } else {
-        throw new ArgumentError('Invalid request body "$body".');
+   if(encoding != null) request.encoding = encoding; else {
```
Comment (natebosch) [inline lib/src/base_client.dart]: We will need to run `dartfmt` and the body of the `if` needs to be wrapped in `{}`.

File: lib/src/request.dart [hunk 2]
```diff
@@ -56,7 +58,10 @@ class Request extends BaseRequest {
     _defaultEncoding = value;
     var contentType = _contentType;
     if (contentType == null) return;
-    _contentType = contentType.change(parameters: {'charset': value.name});
+    if (isIgnoreEncoding)
+      _contentType = contentType.change(parameters: {});
```
Comment (natebosch) [inline lib/src/request.dart]: Any `if` with an `else` needs both blocks in braces.

https://www.dartlang.org/guides/language/effective-dart/style#do-use-curly-braces-for-all-flow-control-structures

**Comment by cskau-g** [comment]
The PR description isn't very descriptive. Can you please edit it to say what this is and why we (all Dart HTTP users) need it?
The two referenced issues seem unrelated?

Have you considered how this will work with Request.encoding?

**Comment by cskau-g** [comment]
Ah, reviewing #184 I think I see that this PR is trying to address that issue specifically?

Parts of this change is a bit misleading since this doesn't really have anything to do with encoding, but is about whether we latch the charset on in Content-Type.
The way I see the problem is that there isn't really a mechanism for controlling the charset directly through the API, and it's automatically latch on where it really should be fully user controlled.
From skimming this, it doesn't really seem like an elegant solution to do that, so I think we might need to put some more thought into how we go about this.

Another aside is that something like this definitely needs tests.
We want to make sure we don't silently break clients who rely on the quirk of charset being set.

**Comment by mit-mit** [comment]
Closing as CLA wasn't signed

**Pull request closed without merging**